packages/readmodel/src/eserviceTemplateReadModelService.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  EServiceTemplate,
+  EServiceTemplateId,
+  WithMetadata,
+} from "pagopa-interop-models";
+import {
+  DrizzleReturnType,
+  eserviceTemplateInReadmodelEserviceTemplate,
+  eserviceTemplateRiskAnalysisAnswerInReadmodelEserviceTemplate,
+  eserviceTemplateRiskAnalysisInReadmodelEserviceTemplate,
+  eserviceTemplateVersionAttributeInReadmodelEserviceTemplate,
+  eserviceTemplateVersionDocumentInReadmodelEserviceTemplate,
+  eserviceTemplateVersionInReadmodelEserviceTemplate,
+  eserviceTemplateVersionInterfaceInReadmodelEserviceTemplate,
+} from "pagopa-interop-readmodel-models";
+import { and, eq, lte } from "drizzle-orm";
+import { splitEServiceTemplateIntoObjectsSQL } from "./eservice-template/splitters.js";
+import {
+  aggregateEServiceTemplate,
+  toEServiceTemplateAggregator,
+} from "./eservice-template/aggregators.js";
+import { checkMetadataVersion } from "./utils.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function eserviceTemplateReadModelServiceBuilder(db: DrizzleReturnType) {
+  return {
+    // eslint-disable-next-line sonarjs/cognitive-complexity
+    async upsertEServiceTemplate(
+      eserviceTemplate: EServiceTemplate,
+      metadataVersion: number
+    ): Promise<void> {
+      await db.transaction(async (tx) => {
+        const shouldUpsert = await checkMetadataVersion(
+          tx,
+          eserviceTemplateInReadmodelEserviceTemplate,
+          metadataVersion,
+          eserviceTemplate.id
+        );
+
+        if (shouldUpsert) {
```

Comment:
minor to reduce cognitive complexity
```suggestion
        if (!shouldUpsert)
          return;
```

Reply:
Thanks. Updated all occurrences.